src/simulation/simulation.jl

```diff
@@ -196,49 +213,99 @@ function _check_folder(sim::Simulation)
     end
 end
 
-function _build_problems!(sim::Simulation, serialize)
-    for (model_number, model) in enumerate(get_models(sim))
-        name = get_name(model)
-        @info("Building problem $(model_number)-$(name)")
+function _build_decision_models!(sim::Simulation)
+    for (model_number, model) in enumerate(get_decision_models(get_models(sim)))
+        @info("Building problem $(get_name(model))")
         initial_time = get_initial_time(sim)
         set_initial_time!(model, initial_time)
         output_dir = joinpath(get_models_dir(sim))
         set_output_dir!(model, output_dir)
-        initialize_simulation_info!(model, problem_chronology)
-        problem_build_status = _build!(model, serialize)
-        if problem_build_status != BuildStatus.BUILT
-            error("Problem $(name) failed to build successfully")
+        try
+            TimerOutputs.@timeit BUILD_PROBLEMS_TIMER "Problem $(get_name(model))" begin
+                # TODO-PJ: Temporary while are able to switch from PJ to POI
+                container = get_optimization_container(model)
+                container.built_for_recurrent_solves = true
+                build_impl!(model)
+            end
+            sim.internal.date_ref[model_number] = initial_time
+            set_status!(model, BuildStatus.BUILT)
+        catch
+            set_status!(model, BuildStatus.FAILED)
+            rethrow()
+        end
+    end
+    return
+end
+
+function _build_emulation_model!(sim::Simulation)
+    model = get_emulation_model(get_models(sim))
+
+    if model === nothing
+        return
+    end
+
+    try
+        get_execution_count(model)
+        TimerOutputs.@timeit BUILD_PROBLEMS_TIMER "Problem $(get_name(model))" begin
+            build_impl!(model)
         end
-        _populate_caches!(sim, name)
         sim.internal.date_ref[model_number] = initial_time
+        set_status!(model, BuildStatus.BUILT)
+    catch
+        set_status!(model, BuildStatus.FAILED)
+        rethrow()
     end
     return
 end
 
+function _initialize_simulation_state(sim::Simulation) end
```

Comment:
Is this a placeholder for future work? Is that listed somewhere?

Reply:
yes, this is a place holder at the moment.